src/Microsoft.Azure.WebJobs.Host/Executors/StorageAccountParser.cs

```diff
@@ -70,6 +71,9 @@ public static StorageAccountParseResult TryParseAccount(string connectionString,
         /// <returns>Formatted error message with details about reason of the failure and possible ways of mitigation</returns>
         public static string FormatParseAccountErrorMessage(StorageAccountParseResult error, string connectionStringName)
         {
+            // Users may accidentally use their real connection strings here, so let's be safe before throwing.
+            connectionStringName = Sanitizer.Sanitize(connectionStringName);
```

Comment:
Exception messages inside of ExceptionTelemetry cannot be sanitized, so handling at the source.